helper/max_length_slice.go

```diff
@@ -0,0 +1,64 @@
+package helper
+
+import "sync"
+
+// MaxLengthStringSliceAdaptor wrap the given slice with several methods for MaxLenSlice.
+// Note that this adaptor does not own that slice at all, so keep the content of slice unchanged!
+type MaxLengthStringSliceAdaptor struct {
+	s      []string
+	maxlen int
+	lock   sync.Mutex
+}
+
+// NewMaxLengthStringSliceAdaptor creates an adaptor for given string slice
+func NewMaxLengthStringSliceAdaptor(s []string, maxlen int) *MaxLengthStringSliceAdaptor {
+	result := &MaxLengthStringSliceAdaptor{
+		s:      s,
+		maxlen: maxlen,
+	}
+	result.removeExceededItems()
+	return result
+}
+
+// MaxLen returns the maximum length of given maxlenslice
+func (m MaxLengthStringSliceAdaptor) MaxLen() int {
+	return m.maxlen
+}
+
+// Use it when you acquire the lock
+func (m *MaxLengthStringSliceAdaptor) removeExceededItems() {
+	if len(m.s) > m.MaxLen() {
+		m.s = m.s[len(m.s)-m.MaxLen():]
```

Comment:
@shanzi This will be leaved as a challenge for freshmen in tomorrow's development :frog: